packages/broker/test/integration/createMessagingPluginTest.ts

```diff
@@ -0,0 +1,124 @@
+import { Stream, StreamrClient } from 'streamr-client'
+import { startTracker, Tracker } from 'streamr-network'
+import { Broker } from '../../src/broker'
+import { Message } from '../helpers/PayloadFormat'
+import { createClient, startBroker, createTestStream, createMockUser, Queue } from '../utils'
+
+interface MessagingPluginApi<T> {
+    createClient: (action: 'publish'|'subscribe', streamId: string, apiKey: string) => Promise<T>
+    closeClient: (client: T) => Promise<void>
+    publish: (message: Message, streamId: string, client: T) => Promise<void>,
+    subscribe: (messageQueue: Queue<Message>, streamId: string, client: T) => Promise<void>
+}
+
+const MOCK_MESSAGE = { 
+    content: { 
+        foo: 'bar' 
+    }, 
+    metadata: {
+        timestamp: 11111111
+    } 
+}
+const MOCK_API_KEY = 'mock-api-key'
+
+const brokerUser = createMockUser()
+
+const assertReceivedMessage = (message: Message) => {
+    const { content, metadata } = message
+    expect(content).toEqual(MOCK_MESSAGE.content)
+    expect(metadata.timestamp).toEqual(MOCK_MESSAGE.metadata.timestamp)
+    expect(metadata.sequenceNumber).toEqual(0)
+    expect(metadata.publisherId).toEqual(brokerUser.address.toLocaleLowerCase())
+    expect(metadata.msgChainId).toBeDefined()
+}
+
+export const createMessagingPluginTest = <T>(
+    pluginName: string, 
+    api: MessagingPluginApi<T>, 
+    pluginPort: number, 
+    testModule: NodeJS.Module,
+    pluginConfig: any = {}
+) => {
+
+    // some free ports
+    const LEGACY_WEBSOCKET_PORT = pluginPort + 1
+    const TRACKER_PORT = pluginPort + 2
+    const NETWORK_PORT = pluginPort + 3
```

Comment:
I'd rather these be given explicitly. Reason: port conflicts in tests are not unusual, and being able to simply grep exact port values is useful in identifying them.

Reply:
Now the port numbers are  listed explicitly